### PR TITLE
Nan issue

### PR DESCRIPTION
**Pull request by ffteja**
PR for issue #4066

### CODE REVIEW

**Comment by pelson** [comment]
:+1:

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -627,14 +627,15 @@ def draw(self, renderer):
...
-
             mtext = self if len(info) == 1 else None
+            
             x = x + posx
             y = y + posy
             if renderer.flipy():
                 y = canvash - y
+            if not np.isfinite(x) or not np.isfinite(y):
+                continue
+
```
Comment (tacaswell) [inline lib/matplotlib/text.py]: whitespace in this line

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -627,14 +627,15 @@ def draw(self, renderer):
         angle = self.get_rotation()
 
         for line, wh, x, y in info:
-            if not np.isfinite(x) or not np.isfinite(y):
-                continue
-
             mtext = self if len(info) == 1 else None
+
```
Comment (tacaswell) [inline lib/matplotlib/text.py]: whitespace in this line

**Comment by tacaswell** [comment]
You have trailing whitespace in two of the blank lines which is what is making travis fail.

**Comment by mdboom** [comment]
:+1:

**Comment by mdboom** [comment]
Hmm... if the mathtext points contain non-finite values, I'd consider that a bug in the mathtext renderer.  `posx` and `posy` could, of course, be non-finite for a number of reasons.  So I would take this check outside of the main loop and check for `posx` or `posy` being non-finite.

File: lib/matplotlib/text.py [hunk 3]
```diff
@@ -627,14 +627,15 @@ def draw(self, renderer):
...
             x = x + posx
             y = y + posy
             if renderer.flipy():
                 y = canvash - y
+            if not np.isfinite(x) or not np.isfinite(y):
```
Comment (tacaswell) [inline lib/matplotlib/text.py]: Can you move this check up to just after line 610?

**Comment by tacaswell** [comment]
@ffteja Can you make the requested change?

**Comment by efiring** [comment]
It looks like this is functionally ready to go.  It would be nice to have it squashed down to a single commit.  A test would also be nice, if a simple one can be constructed.

**Comment by tacaswell** [comment]
This needs a re-base (due to the text wrapping PR).

I am not sure we want to promote this to an exception rather than just not drawing.

**Comment by tacaswell** [comment]
@ffteja Can you rebase this?

**Comment by tacaswell** [comment]
@ffteja I have done the re-base + squash and opened a new PR (#4535)

**Pull request closed without merging**